newrelic/structures_newrelic_nrql_alert_condition.go

```diff
@@ -260,9 +345,37 @@ func flattenNrqlAlertCondition(accountID int, condition *alerts.NrqlAlertConditi
 		return fmt.Errorf("[DEBUG] Error setting nrql alert condition `nrql`: %v", err)
 	}
 
+	// setting terms explicitly, critical/warning are not set
 	configuredTerms := d.Get("term").(*schema.Set).List()
-	if err := d.Set("term", flattenNrqlTerms(condition.Terms, configuredTerms)); err != nil {
-		return fmt.Errorf("[DEBUG] Error setting nrql alert condition `term`: %v", err)
+
+	if len(configuredTerms) > 0 {
+		if err := d.Set("term", flattenNrqlTerms(condition.Terms, configuredTerms)); err != nil {
+			return fmt.Errorf("[DEBUG] Error setting nrql alert condition `term`: %v", err)
+		}
+	} else {
+		// Handle the named condition priorities.
+		conditionTerms := flattenNrqlTerms(condition.Terms, configuredTerms)
+
+		for _, term := range conditionTerms {
+			switch term["priority"].(string) {
+			case "critical":
+				t := term
+				delete(t, "priority")
```

Comment:
Kind of have a similar feeling to this as the other comment. Wondering if this could added to the deprecated scenario vs deleting it from the future facing setup.

Reply:
This one might get a little squirrelly unless we duplicate a considerable chunk code.

Reply:
Fair enough